sky/resources.py

```diff
@@ -945,4 +990,8 @@ def __setstate__(self, state):
         if isinstance(image_id, str):
             state['_image_id'] = {state.get('_region', None): image_id}
 
+        disk_type = state.get('_disk_type', None)
+        if isinstance(disk_type, str):
+            state['_disk_type'] = disk_type
+
```

Comment:
```suggestion
        if version < 9:
            self._disk_type = None
```

Also, increase the `Resources.Version`